metric_test.go

```diff
@@ -20,3 +20,27 @@ func BenchmarkSetId(b *testing.B) {
 		metric.SetId()
 	}
 }
+
+func TestTagValidation(t *testing.T) {
+	type testCase struct {
+		tag       []string
+		expecting bool
+	}
+
+	testCases := []testCase{
+		{[]string{"abc=cba"}, true},
+		{[]string{"a="}, true},
```

Comment:
this is a tag expression. not a tag. why is this allowed? i thought we would only allow `k=v` format